plugins/kinetic-dp/fu-kinetic-dp-secure-aux-isp.c

```diff
@@ -0,0 +1,1238 @@
+		error)) {
+		g_prefix_error(error, "failed to send App init data: ");
+		return FALSE;
+	}
+	fu_progress_step_done(progress);
+	g_debug("sending CMDB Block...");
+	if (fu_kinetic_dp_firmware_get_cmdb_block_size(firmware)) {
+		/* Send CMDB */
+		ptr = (guint8 *)fw_data + SPI_CMDB_BLOCK_START;
+		if (!fu_kinetic_dp_secure_aux_isp_send_payload(
+			self,
+			device,
+			connection,
+			ptr,
+			fu_kinetic_dp_firmware_get_cmdb_block_size(firmware),
+			fu_progress_get_child(progress),
+			10000,
+			200,
+			error)) {
+			g_prefix_error(error, "failed to send CMDB: ");
+			return FALSE;
+		}
+	}
+	fu_progress_step_done(progress);
+
+	/* Send Application Identifier */
+	g_debug("sending App ID data...");
+	ptr = (guint8 *)fw_data + SPI_APP_ID_DATA_START;
+	if (!fu_kinetic_dp_secure_aux_isp_send_payload(self,
+						       device,
+						       connection,
+						       ptr,
+						       STD_APP_ID_SIZE,
+						       fu_progress_get_child(progress),
+						       10000,
+						       200,
+						       error)) {
+		g_prefix_error(error, "failed to send App ID data: ");
+		return FALSE;
+	}
+	fu_progress_step_done(progress);
+
+	return TRUE;
+}
+
+static gboolean
+fu_kinetic_dp_secure_aux_isp_install_fw_images(FuKineticDpSecureAuxIsp *self,
+					       FuKineticDpDevice *device,
+					       FuKineticDpConnection *connection,
+					       GError **error)
+{
+	guint8 cmd_id = KT_DPCD_CMD_INSTALL_IMAGES;
+	guint8 status;
+	guint16 wait_count = 1500;
+	g_debug("programming firmware...");
+	if (!fu_kinetic_dp_secure_aux_isp_write_kt_prop_cmd(connection, cmd_id, error)) {
+		g_prefix_error(error, "failed to send DPCD command: ");
+		return FALSE;
+	}
+	while (wait_count-- > 0) {
+		if (!fu_kinetic_dp_connection_read(connection,
+						   DPCD_ADDR_CMD_STATUS_REG,
+						   &status,
+						   1,
+						   error)) {
+			g_prefix_error(error, "failed to read DPCD_MCA_CMD_REG: ");
+			return FALSE;
+		}
+
+		/* target responded */
+		if (status != (cmd_id | DPCD_KT_CONFIRMATION_BIT)) {
+			/* confirmation bit is cleared */
+			if (status == cmd_id) {
+				g_debug("programming firmware done");
+				return TRUE;
+			} else {
+				g_set_error_literal(error,
+						    FWUPD_ERROR,
+						    FWUPD_ERROR_INTERNAL,
+						    "failed to install images");
+				return FALSE;
+			}
+		}
+
+		/* Wait 50ms */
+		g_usleep(INSTALL_IMAGE_POLL_INTERVAL_MS * 1000);
+	}
+	g_set_error_literal(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "installing images timed-out");
+	return FALSE;
+}
+
+static void
+fu_kinetic_dp_secure_aux_isp_send_reset_command(FuKineticDpConnection *connection)
+{
+	g_autoptr(GError) error_local = NULL;
+	if (!fu_kinetic_dp_secure_aux_isp_write_kt_prop_cmd(connection,
+							    KT_DPCD_CMD_RESET_SYSTEM,
+							    &error_local))
+		g_warning("failed to reset system: %s", error_local->message);
+}
+
+static KtFlashBankIdx
+fu_kinetic_dp_secure_aux_isp_get_flash_bank_idx(FuKineticDpConnection *connection, GError **error)
```

Comment:
It would be more gobject to have:


    static gboolean
    fu_kinetic_dp_secure_aux_isp_get_flash_bank_idx(FuKineticDpConnection *connection, KtFlashBankIdx *udx_out, GError **error)

Reply:
fixed